backend/app.hopps.zugferd/src/main/java/app/hopps/zugferd/KafkaConnector.java

```diff
@@ -1,34 +1,54 @@
 package app.hopps.zugferd;
 
-import app.hopps.zugferd.model.InvoiceData;
-import app.hopps.zugferd.model.InvoiceDocument;
+import app.hopps.commons.DocumentData;
+import app.hopps.commons.InvoiceData;
+import app.hopps.zugferd.auth.FinRestClientImpl;
+import io.quarkus.oidc.client.OidcClient;
+import io.quarkus.oidc.client.runtime.TokensHelper;
 import jakarta.enterprise.context.ApplicationScoped;
 import jakarta.inject.Inject;
+import org.eclipse.microprofile.reactive.messaging.Channel;
+import org.eclipse.microprofile.reactive.messaging.Emitter;
 import org.eclipse.microprofile.reactive.messaging.Incoming;
-import org.eclipse.microprofile.reactive.messaging.Outgoing;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.time.Duration;
+
 @ApplicationScoped
 public class KafkaConnector {
+    private static final Logger LOG = LoggerFactory.getLogger(KafkaConnector.class);
 
-    @Inject
-    ZugFerdService zugFerdService;
+    private final ZugFerdService zugFerdService;
+    private final Emitter<InvoiceData> emitter;
+    private final OidcClient oidcClient;
+    private final TokensHelper tokensHelper = new TokensHelper();
 
     @Inject
-    DocumentDownloader documentDownloader;
+    public KafkaConnector(ZugFerdService zugFerdService, @Channel("document-data-out") Emitter<InvoiceData> emitter,
+            OidcClient oidcClient) {
+        this.zugFerdService = zugFerdService;
+        this.emitter = emitter;
+        this.oidcClient = oidcClient;
+    }
 
-    private final Logger LOG = LoggerFactory.getLogger(KafkaConnector.class);
+    @Incoming("document-data-in")
+    public void process(DocumentData documentData) throws URISyntaxException {
+        LOG.info("Received new document data");
+        FinRestClientImpl restClient = new FinRestClientImpl(documentData.internalFinUrl());
 
-    @Incoming("document-uri-in")
-    @Outgoing("document-data-out")
-    public InvoiceData process(InvoiceDocument invoiceDocument) {
+        String accessToken = tokensHelper.getTokens(oidcClient).await().atMost(Duration.ofSeconds(3)).getAccessToken();
         // Process the incoming message payload and return an updated payload
-        try {
-            return zugFerdService.scanInvoice(documentDownloader.downloadDocument(invoiceDocument.URI()));
+        try (InputStream invoice = restClient.getDocument(accessToken)) {
+            LOG.info("Document successful downloaded");
+            InvoiceData invoiceData = zugFerdService.scanInvoice(documentData.referenceKey(), invoice);
+            LOG.info("Invoice scanned");
```

Comment:
Ich hätte hier noch den referencekey o.ä. in die logs getan, damit man sehen kann welche invoices gesscannt wurden.